core/controllers/profile.py

```diff
@@ -134,71 +149,187 @@ def post(self, _):
         self.render_json({})
 
 
-class PreferencesHandler(base.BaseHandler):
-    """Provides data for the preferences page."""
+class ProfilePictureDataUrlHandler(base.BaseHandler):
+    """Provides data url for profile picture."""
+
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {},
+        'PUT': {
+            'update_type': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
```

Comment:
You no longer need this, since this handle only handles the profile picture.

Reply:
As I am not working on that issue, I have removed this schema.